### PR TITLE
Feature/s3 url

### PR DESCRIPTION
**Pull request by daiLlew**
### What

Changes to read & pass on the S3 URL from the inbound kafka message. Removed AWS config from env vars - this is now derived from the S3 URL.

### How to review

Get my dp-dd-file-uploader pull request [https://github.com/ONSdigital/dp-dd-file-uploader/pull/6](url)
The splitter should get the file from the s3URL, process it and then include the s3URL in all outbound messages.

### Who can review

@CarlHembrough @ian-kent @NeilMadden @cmcgarel @fenallen

### CODE REVIEW

File: message/event/uploadEvent.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	Time  int64  `json:"time"`
+	S3URL string `json:"s3URL"`
+}
+
+func (event *UploadEvent) ConvertToEventDetails() (*EventDetails, error) {
+	s3url, err := url.Parse(event.S3URL)
+	if err != nil {
+		return nil, err
+	}
+	_, err = url.ParseRequestURI(s3url.RequestURI())
```
Comment (ian-kent) [inline message/event/uploadEvent.go]: not required afaik, already done by `url.Parse`

File: message/event/uploadEvent.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+// FileUploaded event
+type EventDetails struct {
+	Time       int64
+	FilePath   string
+	BucketName string
+	S3URL      string
+}
+
+func (e *EventDetails) String() string {
```
Comment (ian-kent) [inline message/event/uploadEvent.go]: not sure what this function is for (can't immediately see it being used anywhere) but it feels wrong and adds a json dependency to this package

File: message/event/uploadEvent.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	"net/url"
+	"strings"
+)
+
+type UploadEvent struct {
+	Time  int64  `json:"time"`
+	S3URL string `json:"s3URL"`
+}
+
+func (event *UploadEvent) ConvertToEventDetails() (*EventDetails, error) {
```
Comment (ian-kent) [inline message/event/uploadEvent.go]: any reason for creating a struct then calling ConvertToEventDetails and returning another struct?

could just have `Create(s3URL string, t time.Time) (*EventDetails, error)` and avoid a struct

File: main.go [hunk 4]
```diff
@@ -3,7 +3,7 @@ package main
 import (
 	"github.com/ONSdigital/dp-csv-splitter/aws"
 	"github.com/ONSdigital/dp-csv-splitter/config"
-	"github.com/ONSdigital/dp-csv-splitter/handlers"
+	//"github.com/ONSdigital/dp-csv-splitter/handlers"
```
Comment (ian-kent) [inline main.go]: should remove if not required

**Review by ian-kent — CHANGES_REQUESTED** [review]

File: message/event/uploadEvent.go (hunk 3 above)
Comment (daiLlew) [reply message/event/uploadEvent.go]: The kafka message is converted to the struct type with json.Marshall - as a result it won't convert the S3URL to a URL - having it as a URL and not a string is required to extract some of the other fields. I added this convert so the model that gets passed around is in correct state.

File: message/event/uploadEvent.go (hunk 1 above)
Comment (daiLlew) [reply message/event/uploadEvent.go]: Parse string to url didn't actually pick up an invalid value in my test.

Comment (ian-kent) [reply message/event/uploadEvent.go]: it should, it does the same thing internally:
https://golang.org/src/net/url/url.go?s=11553:11592#L408

File: message/event/uploadEvent.go (hunk 2 above)
Comment (daiLlew) [reply message/event/uploadEvent.go]: Re implemented this with a custom Json unmarshaller as discussed.

**Review by ian-kent — APPROVED** [review]

**Pull request merged**